src/socket/unix_socket.cr

```diff
@@ -68,27 +68,8 @@ class UNIXSocket < Socket
   # left.gets # => "message"
   # ```
   def self.pair(type : Type = Type::STREAM) : {UNIXSocket, UNIXSocket}
-    {% if flag?(:wasm32) || flag?(:win32) %}
-      raise NotImplementedError.new "UNIXSocket.pair"
-    {% else %}
-      fds = uninitialized Int32[2]
-
-      socktype = type.value
-      {% if LibC.has_constant?(:SOCK_CLOEXEC) %}
-        socktype |= LibC::SOCK_CLOEXEC
-      {% end %}
-
-      if LibC.socketpair(Family::UNIX, socktype, 0, fds) != 0
-        raise Socket::Error.new("socketpair() failed")
-      end
-
-      {% unless LibC.has_constant?(:SOCK_CLOEXEC) %}
-        Crystal::System::Socket.fcntl(fds[0], LibC::F_SETFD, LibC::FD_CLOEXEC)
-        Crystal::System::Socket.fcntl(fds[1], LibC::F_SETFD, LibC::FD_CLOEXEC)
-      {% end %}
-
-      {UNIXSocket.new(fd: fds[0], type: type), UNIXSocket.new(fd: fds[1], type: type)}
-    {% end %}
+    fds = Crystal::System::Socket.socketpair(type, Protocol::IP)
+    {UNIXSocket.new(fd: fds[0], type: type), UNIXSocket.new(fd: fds[1], type: type)}
```

Comment:
**minor**: suggestion

```suggestion
    fds = Crystal::System::Socket.socketpair(type, Protocol::IP)
    fds.map { |fd| UNIXSocket.new(fd: fd, type: type) }
```

Reply:
I think I'd prefer the original. The difference in line length isn't that dramatic, yet it's less explicit about what's happening.
This is about _two_ specific sockets, not a generic collection of sockets.

Reply:
Hum, maybe we could go more functional with chaining?

```crystal
Crystal::System::Socket
  .socketpair(type, Protocol::IP)
  .map { |fd| UNIXSocket.new(fd, type) }
```

Reply:
I dig it! Reminds me of functional aspect of Elixir, sigh...